packages/itinerary-body/src/AccessLegBody/rented-vehicle-subheader.js

```diff
@@ -0,0 +1,76 @@
+import { configType, legType } from "@opentripplanner/core-utils/lib/types";
+import {
+  getCompaniesLabelFromNetworks,
+  getModeForPlace
+} from "@opentripplanner/core-utils/lib/itinerary";
+import React from "react";
+
+import * as Styled from "../styled";
+
+/**
+ * A component to display vehicle rental data. The word "Vehicle" has been used
+ * because a future refactor is intended to combine car rental, bike rental
+ * and micromobility rental all within this component. The future refactor is
+ * assuming that the leg.rentedCar and leg.rentedBike response elements from OTP
+ * will eventually be merged into the leg.rentedVehicle element.
+ */
+export default function RentedVehicleSubheader({ config, leg }) {
+  const configCompanies = config.companies || [];
+
+  // Sometimes rented vehicles can be walked over things like stairs or other
+  // ways that forbid the main mode of travel.
+  if (leg.mode === "WALK") {
+    return (
+      <Styled.PlaceSubheader>
+        <span>Walk vehicle along {leg.from.name}</span>
+      </Styled.PlaceSubheader>
+    );
+  }
+
+  let rentalDescription = "Pick up";
+  if (leg.rentedBike) {
+    // TODO: Special case for TriMet may need to be refactored.
+    rentalDescription += ` shared bike`;
```

Comment:
I would rather not concatenate strings (for future translations, if it gets there).